File: jlpt/native_android/app/src/main/java/com/jlpt/android/domain/model/Enums.kt
package com.jlpt.android.domain.model

enum class JlptLevel {
    N5, N4, N3, N2, N1
}

enum class SkillType {
    VOCABULARY, GRAMMAR, READING, LISTENING
}

enum class ExamType {
    FULL_MOCK, SKILL_PRACTICE
}

enum class AttemptStatus {
    IN_PROGRESS, SUBMITTED
}

enum class StudyItemType {
    VOCABULARY, KANJI, GRAMMAR
}

enum class StudyStatus {
    NEW, LEARNING, MASTERED
}

enum class UserRole {
    USER, ADMIN
}

/** Recall-quality buttons shown on the SRS review screen, mapped to the SM-2 0-5 scale. */
enum class ReviewQuality(val score: Int) {
    AGAIN(0),
    HARD(3),
    GOOD(4),
    EASY(5)
}

File: jlpt/native_ios/Sources/Domain/Models/Enums.swift
import Foundation

enum JlptLevel: String, Codable, CaseIterable, Identifiable {
    case n5 = "N5", n4 = "N4", n3 = "N3", n2 = "N2", n1 = "N1"
    var id: String { rawValue }
}

enum SkillType: String, Codable, CaseIterable, Identifiable {
    case vocabulary = "VOCABULARY"
    case grammar = "GRAMMAR"
    case reading = "READING"
    case listening = "LISTENING"
    var id: String { rawValue }
}

enum ExamType: String, Codable {
    case fullMock = "FULL_MOCK"
    case skillPractice = "SKILL_PRACTICE"
}

enum AttemptStatus: String, Codable {
    case inProgress = "IN_PROGRESS"
    case submitted = "SUBMITTED"
}

enum StudyItemType: String, Codable, CaseIterable, Identifiable {
    case vocabulary = "VOCABULARY"
    case kanji = "KANJI"
    case grammar = "GRAMMAR"
    var id: String { rawValue }
}

enum StudyStatus: String, Codable {
    case new = "NEW"
    case learning = "LEARNING"
    case mastered = "MASTERED"
}

enum UserRole: String, Codable {
    case user = "USER"
    case admin = "ADMIN"
}

/// Recall-quality buttons shown on the SRS review screen, mapped to the SM-2 0-5 scale.
enum ReviewQuality: Int, CaseIterable {
    case again = 0
    case hard = 3
    case good = 4
    case easy = 5

    var label: String {
        switch self {
        case .again: return "Lại"
        case .hard: return "Khó"
        case .good: return "Tốt"
        case .easy: return "Dễ"
        }
    }
}
